### PR TITLE
Revamp compile contract script

### LINKED ISSUE #1873: [Task] Tracking ABIs in the codebase
## Introduction and Background

Contract ABIs are not tracking in the codebase now. We should create an `abis/` folder for keeping those files.

/_Describe the background of this task_/

## Content

+ Tracking contract ABIs in the current codebase.
+ Listing the `solcjs`  in the deps list of the project.
+ Rewrite `contract.sh` in JavaScript.

### PR DESCRIPTION
**Pull request by guanbinrui**
close #1873

### CODE REVIEW

**Comment by septs** [comment]
@guanbinrui add `packages/maskbook/abis` to `.prettierignore`?

File: package.json [hunk 1]
```diff
@@ -213,12 +213,13 @@
...
     "storybook-addon-designs": "^5.4.2",
     "storybook-addon-i18n": "^5.1.13",
+    "ts-generator": "^0.0.8",
     "ts-jest": "26.4.4",
     "ts-loader": "^8.0.11",
     "ts-node": "^9.0.0",
     "ttypescript": "^1.5.12",
     "type-coverage": "^2.14.3",
-    "typechain": "^2.0.0",
+    "typechain": "^2.0.1",
```
Comment (Jack-Works) [inline package.json:222]: Would you like to upgrade to typechain@4?

**Review by Jack-Works — APPROVED** [review]

**Review by septs — CHANGES_REQUESTED** [review]
add `/packages/maskbook/abis` to `/.prettierignore`.

**Comment by guanbinrui** [comment]
> add `/packages/maskbook/abis` to `/.prettierignore`.

We need ABIs to be formatted.

**Review by septs — APPROVED** [review]

**Pull request merged**